cmake/set_compile_flags.cmake

```diff
@@ -1,20 +1,29 @@
-# Logic for handling warnings
 if(AMR_WIND_ENABLE_ALL_WARNINGS)
   # GCC, Clang, and Intel seem to accept these
   list(APPEND AMR_WIND_CXX_FLAGS "-Wall" "-Wextra" "-pedantic")
-  if(NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
-    # ifort doesn't like -Wall
-    list(APPEND AMR_WIND_Fortran_FLAGS "-Wall")
-  else()
+  if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
     # Intel always reports some diagnostics we don't necessarily care about
     list(APPEND AMR_WIND_CXX_FLAGS "-diag-disable:11074,11076,15335")
   endif()
-  if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7.0)
-    # Avoid notes about -faligned-new with GCC > 7
-    list(APPEND AMR_WIND_CXX_FLAGS "-faligned-new")
+  if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR
+     "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR
+     "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
+     if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7.0)
+       list(APPEND AMR_WIND_CXX_FLAGS "-faligned-new"
+                                      "-Wno-sign-compare"
+                                      "-Wunreachable-code"
+                                      "-Wnull-dereference"
+                                      "-Wfloat-conversion"
+                                      "-Wshadow"
+                                      "-Woverloaded-virtual")
+    endif()
   endif()
 endif()
 
+if(AMR_WIND_ENABLE_WERROR)
+  list(APPEND AMR_WIND_CXX_FLAGS "-Werror")
```

Comment:
While we are at, we should also add `-fno-operator-names` here.

Reply:
Ok.